prow/hook/server_test.go

```diff
@@ -334,3 +388,188 @@ func TestNeedDemux(t *testing.T) {
 		})
 	}
 }
+
+type roundTripFunc func(req *http.Request) *http.Response
+
+// RoundTrip .
+func (f roundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
+
+// newTestClient returns *http.Client with Transport replaced to avoid making real calls
+func newTestClient(fn roundTripFunc) *http.Client {
+	return &http.Client{
+		Transport: fn,
+	}
+}
+
+func TestDemuxEvent(t *testing.T) {
```

Comment:
@bartoszmajsak , this test is flaky, see https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/25190/pull-test-infra-unit-test/1491504183361146880

Reply:
Oh, interesting. Thanks for sharing it @chaodaiG.
I ran that multiple times (both here through my subsequent commits as well as locally) and haven't noticed that problem yet. I will check what could be the cause and hopefully send a PR with the fix.

Reply:
I just ran it in a loop and indeed roughly one out of ten times it fails on my machine. I will work on the fix today.

Reply:
@chaodaiG https://github.com/kubernetes/test-infra/pull/25222 should fix it. Thanks again!